CRM/Extension/Manager.php

```diff
@@ -221,6 +221,11 @@ public function install($keys) {
       throw new CRM_Extension_Exception('Cannot install incompatible extension: ' . implode(', ', $incompatible));
     }
 
+    // Store a list of extensions that we're processing. This is referenced in
+    // CRM_Core_ManagedEntities which set entities belonging to these
+    // extensions as is_active.
+    Civi::$statics[__CLASS__]['processing'] = $keys;
+
```

Comment:
You might consider this as a way to make the context into a more visible contract:

```php
// Consume list of ext's

$manager = CRM_Extension_System::singleton()->getManager();
if ($task = $manager->getActiveTask() && $task['action'] == 'install') {
  $moduleIsBeingEnabled = in_array($dao->module, $task['keys']);
}

// Provide list of ext's

class CRM_Extension_Manager {

  protected $activeTasks = [];

  /** @return NULL|array('action' => string, 'keys' => string[]) */
  public function getActiveTask() {return $this->activeTasks[0] ?? NULL; }
  protected function pushActiveTask() { array_unshift($this->activeTasks); };
  protected function popActiveTask() { array_pop($this->activeTasks); };

  public function install($keys) {
    $this->pushActiveTask(['action' => 'install', 'keys' => $keys]);
    $cleanup = CRM_Utils_AutoClean::with([$this, 'popActiveTask']);
    // ... Do the install stuff...
  }

  public function disable($keys) {
    $this->pushActiveTask(['action' => 'disable', 'keys' => $keys]);
    $cleanup = CRM_Utils_AutoClean::with([$this, 'popActiveTask']);
    // ... Do the disable stuff...
  }

  public function uninstall($keys) {
    $this->pushActiveTask(['action' => 'uninstall', 'keys' => $keys]);
    $cleanup = CRM_Utils_AutoClean::with([$this, 'popActiveTask']);
    // ... Do the uninstall stuff...
  }

}
```

Note:

* Good: Clarifies the distinction between the 'install' vs 'disable' vs 'uninstall' operations
* Good: Communicates a contract more clearly; supports docblocks, IDE-based discovery, refactoring
* Bad: More SLOC
* Neutral: Formally, this makes reentrancy more plausible. But I'm not sure we have or need reentrancy.